File: kotlin/algorithm/programmers/42747/42747.kt
class Solution_42747 {
    fun solution(citations: IntArray): Int {
        var answer = 0
        citations.sort()
        val length = citations.size
        for(i in citations.indices)
        {
            if(citations[i]>=length-i) return length-i
        }
        return answer
    }
}

fun main() {
    val sol = Solution_42747()
    print(sol.solution(intArrayOf(9, 7,6, 2, 1)))
}

File: kotlin/algorithm/programmers/42747/42747.py
def solution(citations):
    answer = 0
    citations.sort()
    l = len(citations)

    for i in range(l):
        if(citations[i]>=l-i) : return l-i
    return answer

if __name__ == "__main__":
    print(solution([9, 7, 6, 2, 1]))